### PR TITLE
✨ feat(slash-command): add /gh-start-issue command for automated issue workflow

### LINKED ISSUE #458: Add /gh-start-issue slash command for automated issue workflow
## Description

Create a slash command  that automates the workflow for starting work on a GitHub issue. This command should follow best practices from [Claude Code slash commands documentation](https://code.claude.com/docs/en/slash-commands.md).

## Requirements

The command should:
- [ ] Accept a GitHub issue number as an argument
- [ ] Fetch  to ensure we have the latest changes
- [ ] Create a git worktree from  for the issue branch
- [ ] Update the tmux pane name to match the branch name
- [ ] Get started on the issue (read issue details, set up context)
- [ ] Submit a PR when ready (with proper linking to close the issue)

### Command Structure

Following the pattern from existing commands like :

- **Location**: 
- **Argument**: GitHub issue number (e.g., )
- **Frontmatter**: Include  and  fields
- **Bash execution**: Use  prefix for git operations

### Workflow Steps

1. **Parse and validate issue number**
   - Extract issue number from  or 
   - Validate that the issue exists using 

2. **Fetch and prepare**
   - Fetch : `!git fetch origin main`
   - Get issue details: `!gh issue view $ISSUE_NUM --json title,body`

3. **Create worktree**
   - Generate branch name following convention: `<username>/\<issue_num>-\<issue_slug>`
   - Create worktree: `!git worktree add -b "$BRANCH" "wip/$BRANCH" origin/main`
   - Change to worktree directory

4. **Update tmux pane name**
   - Set tmux pane name to branch name: `!tmux rename-window "$BRANCH"` (or appropriate tmux command)

5. **Get started on issue**
   - Display issue title and description
   - Set up context for working on the issue
   - Provide guidance on next steps

6. **PR submission (when ready)**
   - Stage changes: `!git add -A`
   - Commit with conventional format: `!git commit -m "..."`
   - Create PR with proper linking: `!gh pr create --title "..." --body "Fixes #$ISSUE_NUM"`

## Acceptance Criteria

- [ ] Command accepts issue number as argument
- [ ] Command fetches origin main successfully
- [ ] Command creates worktree with correct branch naming convention
- [ ] Command updates tmux pane name (if tmux is available)
- [ ] Command displays issue context and gets started on work
- [ ] Command can create PR that properly links to and closes the issue
- [ ] Command follows project's slash command best practices
- [ ] Command includes proper error handling for missing issues, existing branches, etc.

## References

- [Slash Commands Documentation](https://code.claude.com/docs/en/slash-commands.md)
- Existing command example: 
- Branch naming convention: 
- Conventional commits: 

## Notes

- The command should be small and focused (as requested: "very small slash command")
- Should handle edge cases gracefully (issue doesn't exist, branch already exists, etc.)
- Should work in both tmux and non-tmux environments (tmux commands should be conditional)
- Consider using positional arguments ($1) for the issue number for clarity

### PR DESCRIPTION
**Pull request by codekiln**
## Summary
- Adds `/gh-start-issue` slash command to automate the workflow for starting work on GitHub issues
- Integrates with existing git-worktrees skill and project conventions
- Provides non-interactive, automated setup for issue-based development

## Changes
- Created `.claude/commands/gh-start-issue.md` with complete command implementation
- Validates issue number and fetches issue details
- Checks for parent issues to determine correct target branch
- Creates git worktree from main with proper branch naming convention
- Updates tmux window name if in tmux session  
- Displays issue context and provides next steps guidance
- Includes comprehensive error handling for edge cases

## Related Issues
Fixes #458

## Test Plan
- [x] Command file follows existing slash command patterns
- [x] Validates issue number format
- [x] Handles parent issue detection
- [x] Creates worktree with correct naming
- [x] Updates tmux window when available
- [x] Non-interactive design suitable for automation
- [x] Error handling for missing issues, existing branches, etc.

---
🤖 Generated with [Claude Code](https://claude.com/claude-code)

Co-Authored-By: Claude <noreply@anthropic.com>

### CODE REVIEW

File: .claude/commands/gh-start-issue.md [hunk 1]
```diff
@@ -0,0 +1,201 @@
+---
+description: Create a git worktree from main for a GitHub issue branch and get started on the work
```
Comment (codekiln) [inline .claude/commands/gh-start-issue.md]: follow best practices for creating claude code slash commands https://code.claude.com/docs/en/slash-commands#frontmatter include argument-hint, allowed-tools

File: .claude/commands/gh-start-issue.md [hunk 2]
```diff
@@ -0,0 +1,201 @@
...
+
+```bash
+!echo ""
+echo "━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━"
+echo "📋 Issue #$ISSUE_NUM: $ISSUE_TITLE"
+echo "━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━"
+echo ""
+
+# Display issue body
+echo "$ISSUE_DATA" | jq -r .body
```
[the remark this answers is not in the document]
Comment (codekiln) [reply .claude/commands/gh-start-issue.md]: Fixed in commit 7c3099f: Added null handling with fallback message '(No description provided)' for issues without descriptions.

File: .claude/commands/gh-start-issue.md [hunk 3]
```diff
@@ -0,0 +1,201 @@
...
+if [ -d "$WORKTREE_PATH" ]; then
+  echo "❌ Error: Worktree already exists at $WORKTREE_PATH"
+  echo "Remove it first with: git worktree remove $WORKTREE_PATH"
+  exit 1
+fi
+
+# Check if branch already exists locally or remotely
+if git show-ref --verify --quiet "refs/heads/$BRANCH_NAME" || git ls-remote --heads origin "$BRANCH_NAME" | grep -q "$BRANCH_NAME"; then
+  echo "❌ Error: Branch $BRANCH_NAME already exists"
+  echo "Delete it first with: git branch -D $BRANCH_NAME"
```
[the remark this answers is not in the document]
Comment (codekiln) [reply .claude/commands/gh-start-issue.md]: Fixed in commit 7c3099f: Enhanced error message to provide instructions for deleting both local and remote branches.
